### DISCUSSION TITLE
recorder improvement

### DISCUSSION THREAD

**Discussion by bobzer** [frenck/spook#294] [Ideas] [answered]
Hi,

I don't know if I here is a good place or maybe home assistant directly ? maybe the month of WTH ?
anyway it would be great to :
- be able to rename an entity in the history. (so when i rename an entity I don't lost the history for that entity)
- be able to delete history for a specific entity 
- ask the recorder if this entity is record or not, ideally show with an icon in the vien of the device/entity that this is not saved

have a great day

**Comment by frenck** [comment]
> be able to rename an entity in the history. (so when i rename an entity I don't lost the history for that entity)

That isn't a concern? If you rename an entity, the history will also be renamed. If not, this would be a bug and should be reported.

> be able to delete history for a specific entity

You can already? `recorder.purge_entities`: <https://www.home-assistant.io/integrations/recorder#service-purge_entities>

> ask the recorder if this entity is record or not, ideally show with an icon in the vien of the device/entity that this is not saved

We could make a service to check. This does require that I finish #295 first :)

../Frenck

**Comment by bobzer** [comment]
> > be able to rename an entity in the history. (so when i rename an entity I don't lost the history for that entity)
> 
> That isn't a concern? If you rename an entity, the history will also be renamed. If not, this would be a bug and should be reported.

Is it also true when the entity come from a device that I rename from zigbee2mqtt ? If so I guess I have a bug to report, make sur I test properly before but it happend to me this week (lost th history because of a rename)

> > be able to delete history for a specific entity
> 
> You can already? `recorder.purge_entities`: https://www.home-assistant.io/integrations/recorder#service-purge_entities

sorry my mistake, thank you 👍 

> > ask the recorder if this entity is record or not, ideally show with an icon in the vien of the device/entity that this is not saved
> 
> We could make a service to check. This does require that I finish #295 first :)

indeed that's a great feature but you're too fast it's already merged 🥇 
I guess it could allow to ask for the list of entity corresponding to a glob domain, exclude, include. I mean calling a service with the config that we give to recorder or influxdb for instance and it return the list of entity.

> ../Frenck

keep being as amazing as you're already 👍 

Thank you

**Reply by frenck** [reply, marked as answer]
> Is it also true when the entity come from a device that I rename from zigbee2mqtt ? 

Not sure, I believe Z2M does kinda a weird workaround there to re-add the entity or something to force it.

**Reply by bobzer** [reply]
Great thanks, good to know,
